File: valid-palindrome/valid-palindrome.java
class Solution {
     public boolean isPalindrome(String s) {
       char arr[]=s.toLowerCase().toCharArray();
        StringBuffer str=new StringBuffer();
        for(int i=0;i<arr.length;i++) {
        	if(arr[i]>='a'&&arr[i]<='z'||arr[i]>='0'&&arr[i]<='9')
        		str.append(arr[i]);
        }
        	return str.toString().equals(str.reverse().toString());
    }

}

File: 125-valid-palindrome/valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int l = 0;
        int r = s.length() - 1;

        while (l < r) {
            while (l < r && !isalnum(s[l])) {
                l++;
            }
            while (r > l && !isalnum(s[r])) {
                r--;
            }
            if (l > r || tolower(s[l]) != tolower(s[r])) {
                return false;
            } else {
                l++;
                r--;
            }
        }
        return true;
    }
};